Don't create duplicate issues when configuration is invalid

csomh opened the pull request:
Kudos to @ppisar for reporting this :man_singer: 

Signed-off-by: Hunor Csomortáni <csomh@redhat.com>

jpopelka reviewed (approved):
good catch!

csomh commented:
Please hold the horses! I would like to add some tests.

mfocko reviewed (approved):
damn that suffix idea... :upside_down_face:

csomh commented:
One more round of review please! Here is what changed:
- Changed the way `create_issue_if_needed` works. There is a more explicit `title` arg now, which should simplify things and the only thing that is added in the method is a `[packit] ` prefix.
- Added a few unit tests for this method, just to be sure :stuck_out_tongue_winking_eye:

mfocko commented on tests/unit/test_config.py:
```diff
@@ -335,3 +335,49 @@ def test_get_package_config_from_repo_not_found_exception_nonexisting_issue():
+    if create_new:
+        project.should_receive("create_issue").with_args(
+            title=f"[packit] {title}", body=message
+        ).and_return(flexmock(name="new issue")).once()
+        check = lambda value: value.name == "new issue"  # noqa
```
```suggestion
        ).and_return(flexmock(title="new issue")).once()
        check = lambda value: value.title == "new issue"  # noqa
```

mfocko reviewed (approved):
LGTM :+1:

lachmanfrantisek commented:
Thanks!
